src/app/shared/services/firebase.service.ts

```diff
@@ -1,9 +1,231 @@
+        : this.firestore.collection<Medicine>('medicines');
+
+    this.medicines = this.medicinesCollection.snapshotChanges().pipe(
+        map(actions => actions.map( a =>{
+              let data = a.payload.doc.data() as Medicine;
+              data.id = a.payload.doc.id;
+              return data;
+            }
+        ))
+    );
+
+    return this.medicines;
+  }
+
+  updateMedicine(medicine: Medicine){
+    return this.firestore.doc("medicines/"+medicine.id).update({...medicine});
+  }
+
+  deleteMedicine(medicineId: string){
+    return this.firestore.doc("medicines/" + medicineId).delete();
+  }
+
+  // --------------------------------------------------------------------------
+
+  // ------------------------------ Couriers ----------------------------------
+
+  couriersCollection: AngularFirestoreCollection<Courier>;
+  couriers: Observable<Courier[]>;
+
+  createCourier(courier: Courier): Promise<DocumentReference>{
+    return this.firestore.collection('couriers').add({...courier});
+  }
+
+  getCouriers(queryFn?: QueryFn){
+
+    this.couriersCollection = queryFn
+        ? this.firestore.collection<Courier>('couriers', queryFn)
+        : this.firestore.collection<Courier>('couriers');
+
+    this.couriers = this.couriersCollection.snapshotChanges().pipe(
+        map(actions => actions.map( a =>{
+              let data = a.payload.doc.data() as Courier;
+              data.id = a.payload.doc.id;
+              return data;
+            }
+        ))
+    );
+
+    return this.couriers;
+  }
+
+  updateCourier(courier: Courier){
+    return this.firestore.doc("couriers/"+courier.id).update({...courier});
+  }
+
+  deleteCourier(courierId: string){
+    return this.firestore.doc("couriers/" + courierId).delete();
+  }
+
+  // --------------------------------------------------------------------------
+
+  // ------------------------------ Orders ----------------------------------
+
+  ordersCollection: AngularFirestoreCollection<Order>;
+  orders: Observable<Order[]>;
+
+  createOrder(order: Order): Promise<DocumentReference>{
+    return this.firestore.collection('orders').add({...order});
+  }
+
+  getOrders(queryFn?: QueryFn){
+
+    this.ordersCollection = queryFn
+        ? this.firestore.collection<Order>('orders', queryFn)
+        : this.firestore.collection<Order>('orders');
+
+    this.orders = this.ordersCollection.snapshotChanges().pipe(
+        map(actions => actions.map( a =>{
+              let data = a.payload.doc.data() as Order;
+              data.id = a.payload.doc.id;
+              return data;
+            }
+        ))
+    );
+
+    return this.orders;
+  }
+
+  updateOrder(order: Order){
+    return this.firestore.doc("orders/"+order.id).update({...order});
+  }
+
+  deleteOrder(orderId: string){
+    return this.firestore.doc("orders/" + orderId).delete();
+  }
+
+  // --------------------------------------------------------------------------
+
```

Comment:
Too long file, please split in orders.service.ts, patient.service.ts, couriers.service.ts

Reply:
Done